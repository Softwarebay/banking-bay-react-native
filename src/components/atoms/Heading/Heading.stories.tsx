import { color, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { colors } from 'theme';
import { ViewStory } from 'utils/storybook/ViewStory';

import { Heading } from './Heading';

storiesOf('Atoms/Heading', module)
	.addDecorator((getStory) => <ViewStory center>{getStory()}</ViewStory>)
	.add('h1', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h1"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h2', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h2"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h3', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h3"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h4', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h4"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h5', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h5"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h6', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h6"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h3 success', () => (
		<Heading
			color={color('color', colors.green)}
			size="h3"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h3 error', () => (
		<Heading
			color={color('color', colors.pink)}
			size="h3"
			testID={text('testID', 'Heading-test-id')}
			text={text('title', 'Heading')}
		/>
	))
	.add('h3 long text', () => (
		<Heading
			color={color('color', colors.mainDark)}
			size="h3"
			testID={text('testID', 'Heading-test-id')}
			text={text(
				'title',
				'Lorem ipsum dolor sit amet, consectetur adipiscing elit'
			)}
		/>
	));
