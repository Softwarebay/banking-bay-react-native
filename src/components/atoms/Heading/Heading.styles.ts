import { StyleSheet } from 'react-native';

import { colors } from 'theme';

export const styles = StyleSheet.create({
	default: {
		color: colors.mainDark,
	},
	h1: {
		fontSize: 44,
		fontWeight: '700',
		lineHeight: 48.4,
	},
	h2: {
		fontSize: 36,
		fontWeight: '700',
		lineHeight: 39.6,
	},
	h3: {
		fontSize: 28,
		fontWeight: '700',
		lineHeight: 30.8,
	},
	h4: {
		fontSize: 20,
		fontWeight: '500',
		lineHeight: 24,
	},
	h5: {
		fontSize: 16,
		fontWeight: '600',
		lineHeight: 20.8,
	},
	h6: {
		fontSize: 14,
		fontWeight: '600',
		lineHeight: 22.4,
	},
});
