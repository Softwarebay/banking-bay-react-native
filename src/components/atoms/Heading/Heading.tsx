import { StyleProp, Text, TextStyle } from 'react-native';

import { styles } from './Heading.styles';

export interface Props {
	text: string;
	size: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
	color?: string;
	style?: StyleProp<TextStyle>;
	testID?: string;
}

export const Heading = ({ text, size, color, style, testID }: Props) => {
	return (
		<Text
			style={[color ? { color } : styles.default, styles[size], style]}
			testID={testID}
		>
			{text}
		</Text>
	);
};
