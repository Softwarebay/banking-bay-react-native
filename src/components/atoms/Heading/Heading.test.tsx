import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { Heading } from './Heading';

import type { Props } from './Heading';

describe('Heading', () => {
	const props: Omit<Props, 'size'> = {
		text: 'Heading',
		testID: 'heading-test-id',
		style: {
			marginBottom: 20,
		},
	};

	it('should match snapshot - h1', () => {
		const tree = testRenderer.create(<Heading {...props} size="h1" />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - h2', () => {
		const tree = testRenderer.create(<Heading {...props} size="h2" />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - h3', () => {
		const tree = testRenderer.create(<Heading {...props} size="h3" />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - h4', () => {
		const tree = testRenderer.create(<Heading {...props} size="h4" />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - h5', () => {
		const tree = testRenderer.create(<Heading {...props} size="h5" />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - h6', () => {
		const tree = testRenderer.create(<Heading {...props} size="h6" />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render heading of size h1', () => {
		render(<Heading {...props} size="h1" />);

		const heading = screen.getByTestId('heading-test-id');

		expect(heading).toHaveTextContent('Heading');
		expect(heading).toHaveStyle({
			fontFamily: 'MulishBold',
			fontSize: 51,
			lineHeight: 56.1,
		});
	});

	it('should render heading of size h2', () => {
		render(<Heading {...props} size="h2" />);

		const heading = screen.getByTestId('heading-test-id');

		expect(heading).toHaveTextContent('Heading');
		expect(heading).toHaveStyle({
			fontFamily: 'MulishBold',
			fontSize: 44,
			lineHeight: 48.4,
		});
	});

	it('should render heading of size h3', () => {
		render(<Heading {...props} size="h3" />);

		const heading = screen.getByTestId('heading-test-id');

		expect(heading).toHaveTextContent('Heading');
		expect(heading).toHaveStyle({
			fontFamily: 'MulishBold',
			fontSize: 33,
			lineHeight: 36.3,
		});
	});

	it('should render heading of size h4', () => {
		render(<Heading {...props} size="h4" />);

		const heading = screen.getByTestId('heading-test-id');

		expect(heading).toHaveTextContent('Heading');
		expect(heading).toHaveStyle({
			fontFamily: 'MulishMedium',
			fontSize: 22,
			lineHeight: 26.4,
		});
	});

	it('should render heading of size h5', () => {
		render(<Heading {...props} size="h5" />);

		const heading = screen.getByTestId('heading-test-id');

		expect(heading).toHaveTextContent('Heading');
		expect(heading).toHaveStyle({
			fontFamily: 'MulishSemiBold',
			fontSize: 17,
			lineHeight: 22.1,
		});
	});

	it('should render heading of size h6', () => {
		render(<Heading {...props} size="h6" />);

		const heading = screen.getByTestId('heading-test-id');

		expect(heading).toHaveTextContent('Heading');
		expect(heading).toHaveStyle({
			fontFamily: 'MulishSemiBold',
			fontSize: 16,
			lineHeight: 25.6,
		});
	});

	it('should apply additional styles from props', () => {
		render(<Heading {...props} size="h6" />);

		const heading = screen.getByTestId('heading-test-id');

		expect(heading).toHaveStyle({
			marginBottom: 20,
		});
	});
});
