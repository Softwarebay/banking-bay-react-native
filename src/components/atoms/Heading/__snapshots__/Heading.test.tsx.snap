// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should match snapshot - h1 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1B1D4D",
      },
      Object {
        "fontFamily": "MulishBold",
        "fontSize": 51,
        "lineHeight": 56.1,
      },
      Object {
        "marginBottom": 20,
      },
    ]
  }
  testID="heading-test-id"
>
  Heading
</Text>
`;

exports[`Heading should match snapshot - h2 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1B1D4D",
      },
      Object {
        "fontFamily": "MulishBold",
        "fontSize": 44,
        "lineHeight": 48.4,
      },
      Object {
        "marginBottom": 20,
      },
    ]
  }
  testID="heading-test-id"
>
  Heading
</Text>
`;

exports[`Heading should match snapshot - h3 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1B1D4D",
      },
      Object {
        "fontFamily": "MulishBold",
        "fontSize": 33,
        "lineHeight": 36.3,
      },
      Object {
        "marginBottom": 20,
      },
    ]
  }
  testID="heading-test-id"
>
  Heading
</Text>
`;

exports[`Heading should match snapshot - h4 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1B1D4D",
      },
      Object {
        "fontFamily": "MulishMedium",
        "fontSize": 22,
        "lineHeight": 26.4,
      },
      Object {
        "marginBottom": 20,
      },
    ]
  }
  testID="heading-test-id"
>
  Heading
</Text>
`;

exports[`Heading should match snapshot - h5 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1B1D4D",
      },
      Object {
        "fontFamily": "MulishSemiBold",
        "fontSize": 17,
        "lineHeight": 22.1,
      },
      Object {
        "marginBottom": 20,
      },
    ]
  }
  testID="heading-test-id"
>
  Heading
</Text>
`;

exports[`Heading should match snapshot - h6 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1B1D4D",
      },
      Object {
        "fontFamily": "MulishSemiBold",
        "fontSize": 16,
        "lineHeight": 25.6,
      },
      Object {
        "marginBottom": 20,
      },
    ]
  }
  testID="heading-test-id"
>
  Heading
</Text>
`;
